assets/js/Script.js

```diff
@@ -0,0 +1,88 @@
+import { Carrito } from "./Carrito.js";
+import { Producto } from "./Producto.js";
+
+// viandas
+const productos = new Producto();
+const productosContainer = document.getElementById("productos");
+
+async function obtenerDatos() {
+  const API_URL = "https://67487dfa5801f51535912443.mockapi.io/Productos";
+  const spinner = document.getElementById("spinner");
+
+  spinner.style.display = "block";
+
+  return fetch(API_URL)
```

Comment:
Usar async/await hace que el código sea más legible y fácil de seguir. Además, se maneja el error apropiadamente y el código no se corta a la mitad.
```suggestion
   try {
     const response = await fetch(API_URL);
     if (!response.ok) {
       throw new Error("Error al obtener los datos");
     }
     return await response.json();
   } catch (error) {
     throw error; // Propaga el error
   } finally {
     spinner.style.display = "none";
   }
```